crates/flux-fhir-analysis/src/conv/mod.rs

```diff
@@ -57,6 +57,7 @@ pub(crate) struct AfterSortck<'a, 'genv, 'tcx> {
     next_type_index: u32,
     next_region_index: u32,
     next_const_index: u32,
+    prim_app_sort: FxHashMap<fhir::BinOp, rty::Sort>,
```

Comment:
Everything around this feels a bit janky to me ... ( the business of figuring out what the output sort of the primitive UIF should be.) Maybe its better to have some "global" knowledge that associates the sort directly with the `BinOp` ?

One thing I dislike is you may want the `property` to talk about multiple values e.g. for sake of example `[>>](x, y) == [<<](x, y)` -- and you cannot now...